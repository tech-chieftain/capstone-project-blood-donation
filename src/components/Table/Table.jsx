import React from 'react';
import { useTranslation } from 'react-i18next';
import { PropTypes } from 'prop-types';
import DonorRow from './DonorRow';

function Table({ donors }) {
  const { t } = useTranslation();
  const StylesClass = {
    grayColor: 'w-full bg-gray-200',
    whiteColor: 'we-full bg-red',
    ceilClass:
      'border px-4 py-2 text-center lg:px-6 mt-4 md:mt-0  whitespace-no-wrap border-b bg-gray',
    headerStyle:
      'border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left',
  };

  return (
    <div>
      <table className="border table-auto  min-w-full">
        <thead>
          <tr className="bg-primary">
            <th className={StylesClass.headerStyle}>{t('table.name')}</th>
            <th className={StylesClass.headerStyle}>{t('table.bloodType')}</th>
            <th className={StylesClass.headerStyle}>{t('table.city')}</th>
            <th className={StylesClass.headerStyle}>{t('table.phone')}</th>
            <th className={StylesClass.headerStyle}>{t('table.email')}</th>
          </tr>
        </thead>
        <tbody> {donors ? donors.map((donor) => <DonorRow donor={donor} />) : null};</tbody>
      </table>
    </div>
  );
}
Table.propTypes = {
  donors: PropTypes,
};
export default Table;
