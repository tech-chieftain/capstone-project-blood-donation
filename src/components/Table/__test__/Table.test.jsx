import React from 'react';
import renderer from 'react-test-renderer';
import Table from '../Table';

it('renders correctly', () => {
  const tree = renderer.create(<Table />).toJSON();
  expect(tree).toMatchSnapshot();
});
