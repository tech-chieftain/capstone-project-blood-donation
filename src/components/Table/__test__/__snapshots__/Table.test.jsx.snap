// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    className="border table-auto  min-w-full"
  >
    <thead>
      <tr
        className="bg-primary"
      >
        <th
          className="border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left"
        >
          table.name
        </th>
        <th
          className="border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left"
        >
          table.bloodType
        </th>
        <th
          className="border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left"
        >
          table.city
        </th>
        <th
          className="border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left"
        >
          table.phone
        </th>
        <th
          className="border px-4 py-2 text-white text-center leading-4 text-sm tracking-wider text-left"
        >
          table.email
        </th>
      </tr>
    </thead>
    <tbody>
       
      ;
    </tbody>
  </table>
</div>
`;
